test/integration/udpingress_test.go

```diff
@@ -23,8 +25,19 @@ import (
 	"github.com/kong/kubernetes-ingress-controller/pkg/clientset"
 )
 
+var udpMutex sync.Mutex
+
 func TestUDPIngressEssentials(t *testing.T) {
 	t.Parallel()
+	// Ensure no other UDP tests run concurrently to avoid fights over the port
+	t.Log("locking UDP port")
+	udpMutex.Lock()
+	defer func() {
+		// Free up the UDP port
+		t.Log("unlocking UDP port")
+		udpMutex.Unlock()
+	}()
+
```

Comment:
Similar to above, I think now might be the time to clean up the technical debt in our tests and make a pool of ports available for TCP and UDP testing, rather than the current static port.

Reply:
moving this conversation to https://github.com/Kong/kubernetes-ingress-controller/pull/1702#discussion_r689675013